userspace/libsinsp/container_engine/docker/async_source.cpp

```diff
@@ -868,6 +867,17 @@ bool docker_async_source::parse(const docker_lookup_request& request, sinsp_cont
 
 	container.m_size_rw_bytes = root["SizeRw"].asInt64();
 
+	try
+	{
+		auto lowerdir = root["GraphDriver"]["Data"]["LowerDir"].asString();
+		container.m_overlayfs_root = lowerdir.substr(lowerdir.find_first_of(':') + 1);
```

Comment:
I'm not 100% sure how this would behave if `':'` is not found in `lowerdir`. My guess is `find_first_of` will return `npos` which is defined as `-1` then `substr` would just copy the entire string since it would resolve to `lowerdir.substr(-1 + 1)`, am I correct?  Are we comfortable relying on this?

Reply:
Exactly this! I think `npos` value is stable (https://cplusplus.com/reference/string/string/npos/) and i do not expect it to change or be arch dependent, therefore imo we can go with this!